projects/core/components/primitive-textfield/primitive-textfield.component.ts

```diff
@@ -205,7 +205,7 @@ export class TuiPrimitiveTextfieldComponent
 
     get placeholderVisible(): boolean {
         const hasDecor =
-            this.nativeFocusableElement?.placeholder || this.prefix || this.postfix;
+            (this.nativeFocusableElement?.placeholder ?? this.prefix) || this.postfix;
```

Comment:
This might also be incorrect, now empty string native placeholder does not fall through.